core/src/main/java/io/questdb/griffin/engine/groupby/YearTimestampSampler.java

```diff
@@ -38,25 +38,42 @@ public class YearTimestampSampler implements TimestampSampler {
     private int startMonth;
     private int startSec;
 
-    public YearTimestampSampler(int bucket) {
-        this.bucket = bucket;
+    public YearTimestampSampler(int stepYears) {
+        this.stepYears = stepYears;
+    }
+
+    @Override
+    public long getApproxBucketSize() {
+        return Timestamps.YEAR_MICROS_NONLEAP * stepYears;
     }
 
     @Override
     public long nextTimestamp(long timestamp) {
-        return addYears(timestamp, bucket);
+        return addYears(timestamp, stepYears);
+    }
+
+    @Override
+    public long nextTimestamp(long timestamp, int numSteps, long maxTimestamp) {
+        long result = timestamp;
+        for (int i = 0; i < numSteps; i++) {
+            result = addYears(result, stepYears);
```

Comment:
@mtopolnik can't we just do `result = addYears(result, numSteps * stepYears);` here?

Reply:
Resolved in 1f8d23d